src/charts/tree/tree.js

```diff
@@ -129,18 +159,25 @@ export default function(DOMNode, options = {}) {
     let nodeIndex = 0
     let maxLabelLength = 0
 
+    // nodes are assigned with string ids, which reflect their location
+    // within the hierarcy; e.g. "root|branch|subBranch|subBranch[0]|property"
+    // top-level elemnt always has id "root"
     visit(data,
-        node => maxLabelLength = Math.max(node.name.length, maxLabelLength),
-        node => node.children && node.children.length > 0 ? node.children : null
+        node => {
+          maxLabelLength = Math.max(node.name.length, maxLabelLength)
+          node.id = node.id || 'root'
+        },
+        node => node.children && node.children.length > 0 ? node.children.map((c) => {
+          c.id = `${node.id || ''}|${c.name}`
+          return c
+        }) : null
     )
 
-    data.x0 = height / 2
-    data.y0 = 0
     /*eslint-disable*/
-    update(data)
+    update()
     /*eslint-enable*/
 
-    function update(source) {
+    function update() {
```

Comment:
No need to pass source any more, because transition positions are now taken from `previousNodePositionsById` / `nodePositionsById`